packages/editor/src/lib/license/Watermark.tsx

```diff
@@ -1,176 +1,170 @@
+import { memo, useState } from 'react'
 import { useCanvasEvents } from '../hooks/useCanvasEvents'
 import { useEditor } from '../hooks/useEditor'
-import { getDefaultCdnBaseUrl } from '../utils/assets'
 import { featureFlags } from '../utils/debug-flags'
 import { stopEventPropagation } from '../utils/dom'
-import { watermarkDesktopSvg } from '../watermarks'
+import { watermarkDesktopSvg, watermarkMobileSvg } from '../watermarks'
 import { LicenseManager } from './LicenseManager'
 import { useLicenseContext } from './LicenseProvider'
 
-/** @internal */
-export const WATERMARK_REMOTE_SRC = `${getDefaultCdnBaseUrl()}/watermarks/watermark-desktop.svg`
-export const WATERMARK_LOCAL_SRC = `data:image/svg+xml;utf8,${encodeURIComponent(watermarkDesktopSvg)}`
-
-let watermarkUrlPromise: Promise<string> | null = null
-async function getWatermarkUrl(forceLocal: boolean): Promise<string> {
-	if (forceLocal) {
-		return WATERMARK_LOCAL_SRC
-	}
-
-	if (!watermarkUrlPromise) {
-		watermarkUrlPromise = Promise.race([
-			// try and load the remote watermark, if it fails, fallback to the local one
-			(async () => {
-				try {
-					const response = await fetch(WATERMARK_REMOTE_SRC)
-					if (!response.ok) return WATERMARK_LOCAL_SRC
-					const blob = await response.blob()
-					return URL.createObjectURL(blob)
-				} catch {
-					return WATERMARK_LOCAL_SRC
-				}
-			})(),
-
-			// but if that's taking a long time (>3s) just show the local one anyway
-			new Promise<string>((resolve) => {
-				// eslint-disable-next-line no-restricted-globals
-				setTimeout(() => {
-					resolve(WATERMARK_LOCAL_SRC)
-				}, 3_000)
-			}),
-		])
-	}
-
-	return watermarkUrlPromise
-}
+const WATERMARK_DESKTOP_LOCAL_SRC = `data:image/svg+xml;utf8,${encodeURIComponent(watermarkDesktopSvg)}`
+const WATERMARK_MOBILE_LOCAL_SRC = `data:image/svg+xml;utf8,${encodeURIComponent(watermarkMobileSvg)}`
 
 /** @internal */
-export const Watermark = React.memo(function Watermark({
-	forceLocal = false,
-}: {
-	forceLocal?: boolean
-}) {
-	const events = useCanvasEvents()
-
-	const editor = useEditor()
+export const Watermark = memo(function Watermark() {
 	const licenseManager = useLicenseContext()
+	const editor = useEditor()
+	const isMobile = useValue('is mobile', () => editor.getViewportScreenBounds().width < 700, [
+		editor,
+	])
+	const [src, setSrc] = useState<string | null>(null)
 
-	const showWatermark = useValue(
-		'show watermark',
-		() =>
-			featureFlags.enableLicensing.get() &&
-			editor.getViewportScreenBounds().width > 760 &&
-			['licensed-with-watermark', 'unlicensed'].includes(licenseManager.state.get()),
-		[editor, licenseManager]
+	useQuickReactor(
+		'set watermark src',
+		async () => {
+			const showWatermark =
+				featureFlags.enableLicensing.get() &&
+				['licensed-with-watermark', 'unlicensed'].includes(licenseManager.state.get())
+
+			if (showWatermark) {
+				setSrc(isMobile ? WATERMARK_MOBILE_LOCAL_SRC : WATERMARK_DESKTOP_LOCAL_SRC)
```

Comment:
Since we removed showing remote watermark images we could also remove uploading unnecessary files to our assets bucket?

Reply:
@SomeHats one for you?

Reply:
Do we support mini? We could add another tiny tiny breakpoint there but it would mean changing our toolbar logic again.

Reply:
360px was my most narrow screen width to support. I think earlier phones had 320px but it's pretty rare to have anything more narrow than 360.

Reply:
> 360px was my most narrow screen width to support. I think earlier phones had 320px but it's pretty rare to have anything more narrow than 360.

iPhone mini is 375px wide.

Reply:
Double checked, it was the iPhone SE that had that tiny 320 screen!